sdk/php/runtime/template/README.md

```diff
@@ -311,3 +311,48 @@ class Example
      // ...
 }
 ```
+
+#### Directories and Files
+
+Directories and Files have some additional metadata that can be specified
+
+##### Defaults
+
+Specifying a default value for a `Directory` or `File` isn't easily possible. 
+Instead you can specify a `DefaultPath` attribute with an absolute or relative path.
+If relative then the path is relative to the `source` specified in the `dagger.json` file.
```

Comment:
Not anymore after https://github.com/dagger/dagger/pull/8391

Reply:
This is my fault for not making it more clear.

What I mean is we can't do a _default value_ in the way we normally can in PHP.

For instance, I cannot do this:

```php
public function myDaggerFunction(
    Directory $dir = '.'
)
```

Because PHP will complain that the string `'.'` is not an instance of the `Directory` object.

Thus we have to use a `DefaultPath` attribute instead.

Thank you for pointing this out, it's very useful to see where I haven't been clear. I'll sort it out shortly!

Reply:
https://github.com/dagger/dagger/blob/d08ed252db2e8bf2fd340450c8cacfbab4a1e1b3/sdk/php/runtime/template/README.md?plain=1#L322-L326

Is this better?

Reply:
It's better yep, I still left a comment to clarify what's the root depending of its relative or absolute path

Reply:
@charjr, the point of the defaultPath isn't to provide a default value for a complex type, in this case `dagger.Directory`. Its purpose is to allow getting files from outside the _source directory_. Without it, the caller of the module needs to explicitly pass in the directory from outside what the source dir is.

Reply:
Note that it only works for directories inside the same git repo (context directory), or within the same module (root directory, i.e., where `dagger.json` is). Before it, modules could only get files from inside of what's in `"source"` (in `dagger.json`).

Reply:
Thank you @helderco I think I have addressed this now.

> Its purpose is to **allow getting files from outside the _source directory_**. 

https://github.com/dagger/dagger/blob/eb1facdfbaf2376efcaf75f2f1bc1870577150b0/sdk/php/runtime/template/README.md?plain=1#L322



> Note that it only works for directories inside the same git repo (context directory), or within the same module (root directory, i.e., where `dagger.json` is) if not in a git repo.

https://github.com/dagger/dagger/blob/eb1facdfbaf2376efcaf75f2f1bc1870577150b0/sdk/php/runtime/template/README.md?plain=1#L323-L325

https://github.com/dagger/dagger/blob/eb1facdfbaf2376efcaf75f2f1bc1870577150b0/sdk/php/runtime/template/README.md?plain=1#L353-L361

Reply:
@helderco @TomChv I think this feedback has been resolved now, please let me know if otherwise. :+1: